### DISCUSSION TITLE
Setting up a development envoirnment

### DISCUSSION THREAD

**Discussion by ParadauxIO** [pterodactyl/panel#3191] [General Community]
Could CONTRIBUTING.md be expanded to show potential contributors how to setup their development envoirnment to best suit quick development? 

I'm not particularly familiar with Laravel or React myself but would like to pitch in, on behalf of my org -- to some of the open issues. 

Thanks.

**Comment by trenutoo** [comment]
You can refer to [BUILDING.md](https://github.com/pterodactyl/panel/blob/develop/BUILDING.md) to assist you with that. Using an existing docker-compose file to spin up the panel in a container is the best method.

**Comment by schrej** [comment]
You can try using [pterodactyl/development](https://github.com/pterodactyl/development). It sets up a dev environment in virtual machines and docker containers. Works on linux and macOS (or at least should work).

The last time I set it up there were some minor issues, not sure if they're fixed now. PRs that fix issues with initial setup are welcome of course. All in all that should be the easiest way to set up a complete dev environment.

**Reply by DaneEveritt** [reply]
Should be good now, I had to setup from scratch the other week.
